File: src/mailmergeandreporting/mailmergeformfields/java/MailMergeFormFields.java
/* 
 * Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Words. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
package mailmergeandreporting.mailmergeformfields.java;

import java.io.File;
import java.net.URI;

import com.aspose.words.*;


/**
 * This sample shows how to insert check boxes and text input form fields during mail merge into a document.
 */
public class MailMergeFormFields
{
    /**
     * The main entry point for the application.
     */
    public static void main(String[] args) throws Exception
    {
        String dataDir = "src/mailmergeandreporting/mailmergeformfields/data/";

        // Load the template document.
        Document doc = new Document(dataDir + "Template.doc");

        // Setup mail merge event handler to do the custom work.
        doc.getMailMerge().setFieldMergingCallback(new HandleMergeField());

        // This is the data for mail merge.
        String[] fieldNames = new String[] {"RecipientName", "SenderName", "FaxNumber", "PhoneNumber",
            "Subject", "Body", "Urgent", "ForReview", "PleaseComment"};
        Object[] fieldValues = new Object[] {"Josh", "Jenny", "123456789", "", "Hello",
            "Test message 1", true, false, true};

        // Execute the mail merge.
        doc.getMailMerge().execute(fieldNames, fieldValues);

        // Save the finished document.
        doc.save(dataDir + "Template Out.doc");
    }

    private static class HandleMergeField implements IFieldMergingCallback
    {
        /**
         * This handler is called for every mail merge field found in the document,
         * for every record found in the data source.
         */
        public void fieldMerging(FieldMergingArgs e) throws Exception
        {
            if (mBuilder == null)
                mBuilder = new DocumentBuilder(e.getDocument());

            // We decided that we want all boolean values to be output as check box form fields.
            if (e.getFieldValue() instanceof Boolean)
            {
                // Move the "cursor" to the current merge field.
                mBuilder.moveToMergeField(e.getFieldName());

                // It is nice to give names to check boxes. Lets generate a name such as MyField21 or so.
                String checkBoxName = java.text.MessageFormat.format("{0}{1}", e.getFieldName(), e.getRecordIndex());

                // Insert a check box.
                mBuilder.insertCheckBox(checkBoxName, (Boolean)e.getFieldValue(), 0);

                // Nothing else to do for this field.
                return;
            }

            // Another example, we want the Subject field to come out as text input form field.
            if ("Subject".equals(e.getFieldName()))
            {
                mBuilder.moveToMergeField(e.getFieldName());
                String textInputName = java.text.MessageFormat.format("{0}{1}", e.getFieldName(), e.getRecordIndex());
                mBuilder.insertTextInput(textInputName, TextFormFieldType.REGULAR, "", (String)e.getFieldValue(), 0);
            }
        }

        public void imageFieldMerging(ImageFieldMergingArgs args) throws Exception
        {
            // Do nothing.
        }

        private DocumentBuilder mBuilder;
    }
}
//ExEnd

File: Plugins/Aspose_Words_for_PHP/src/mailmergeandreporting/mailmergeformfields/php/MailMergeFormFields.php
<?php

/*
 * Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Words. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */

use com\aspose\words\Document as Document;
use com\aspose\words\IFieldMergingCallback as IFieldMergingCallback;
use com\aspose\words\FieldMergingArgs as FieldMergingArgs;
use com\aspose\words\DocumentBuilder as DocumentBuilder;
use com\aspose\words\TextFormFieldType as TextFormFieldType;
use com\aspose\words\ImageFieldMergingArgs as ImageFieldMergingArgs;
use java\lang\Boolean as Boolean;
use java\text\MessageFormat as MessageFormat;


class MailMergeFormFields {

    public static function main()
    {

        $dataDir = "/usr/local/apache-tomcat-8.0.22/webapps/JavaBridge/Aspose_Words_Java_For_PHP/src/mailmergeandreporting/mailmergeformfields/data/";
        $doc = new Document($dataDir . "Template.doc");
        $doc->getMailMerge()->setFieldMergingCallback(new HandleMergeField());

        $fieldNames = array("RecipientName","SenderName","FaxNumber","PhoneNumber","Subject","Body","Urgent","ForReview","PleaseComment");
        $fieldValues = array("Josh","Jenny","123456789","","Hello","Test Pakistan 1", true, false, true);

        $doc->getMailMerge()->execute($fieldNames,$fieldValues);

        $doc->save($dataDir . "Template Out.doc");

    }

}

class HandleMergeField extends IFieldMergingCallback {

    private $mBuilder = null;
    private $mF = null;
    private $tffT = null;

    function __construct()
    {
        $this->mBuilder = new DocumentBuilder();
        $this->mF = new MessageFormat();
        $this->tffT = new TextFormFieldType();
    }

    public function fieldMerging(FieldMergingArgs $e)
    {

        if($this->mBuilder == null)
            $this->mBuilder = new DocumentBuilder($e->getDocument());

        if($e->getFieldValue() instanceof Boolean )
        {
            $this->mBuilder->moveToMergeField($e->getFieldName());
            $checkBoxName = $this->mF->format("{0}{1}",$e->getFieldName(),$e->getRecordIndex());

            $this->mBuilder->insertCheckBox($checkBoxName, (Boolean) $e->getFieldValue(), 0);

            return;
        }

        if(java_values($e->getFieldName()) == 'Subject')
        {
            $this->mBuilder->moveToMergeField($e->getFieldName());
            $textInputName = $this->mF->format("{0}{1}", $e->getFieldName(), $e->getRecordIndex());
            $this->mBuilder->insertTextInput($textInputName, $this->tffT->REGULAR, "", $e->getFieldValue(), 0 );
        }

    }

    public function imageFieldMerging(ImageFieldMergingArgs $args)
    {

    }

}

